joinmarket/taker.py

```diff
@@ -112,14 +112,19 @@ def recv_txio(self, nick, utxo_list, cj_pub, change_addr):
             log.debug(('recv_txio => nick={} not in '
                        'nonrespondants {}').format(nick, self.nonrespondants))
             return
-        self.utxos[nick] = utxo_list
-        utxo_data = jm_single().bc_interface.query_utxo_set(self.utxos[nick])
+        duplicate_utxos = set(sum(self.utxos.values(), [])).intersection(set(utxo_list))
```

Comment:
- `self.utxos.values()` is the sets of coins sent by each nick as inputs
- `sum(list_of_lists, [])` is an idiom for [flatten](https://stackoverflow.com/questions/406121/flattening-a-shallow-list-in-python#comment1198171_408281)